### ISSUE TITLE
backend can't connect to db

### ISSUE THREAD

**Issue by Kevr0n228** [gregordr/ImageStore#33]
![image](https://user-images.githubusercontent.com/10468194/111844110-bcfddf80-88d8-11eb-9b0c-c4617d912b51.png)
As the title says, after running `docker-compose up`, the service starts but the backend can't reach the postgres database. I've left `docker-compose.yml` virtually untouched besides uncommenting the lines for detectron. I'm pretty new to using docker and postgres, so perhaps I'm missing something but I've yet to find it.

`docker-compose.yml`:
```
version: "3.7"
services:
    nginx:
        image: imagestored/nginx
        ports:
            - "3000:8080" #Change to X:8080 to access on port X
        depends_on:
            - frontend

    frontend:
        image: imagestored/frontend
        depends_on:
            - backend

    backend:
        image: imagestored/backend
        restart: always
        environment:
            PGSTRING: postgres://postgres:example@db:5432/postgres # This string has to resolve to a Postgres database on version 11
        volumes:
            - ./media/:/code/media/
        depends_on:
            - db

    db:
        image: postgres:11
        restart: always
        environment:
            POSTGRES_PASSWORD: example
        volumes:
            - ./data/:/var/lib/postgresql/data/

    detectron2: # You can use either detectron2, which is slower but more accurate, or yolo, which is faster but has less accurate results, and is ver>         image: imagestored/detectron2
         environment:
             BASE_ADDRESS: backend:4000 # Set this to the address the backend will be accessible on
             FETCH_INTERVAL: 10 # This is the interval the labeler will poll for new images in seconds

    # yolo: # You can use either detectron2, which is slower but more accurate, or yolo, which is faster but has less accurate results, and is very le>    #     image: imagestored/yolo
    #     environment:
    #         BASE_ADDRESS: backend:4000 # Set this to the address the backend will be accessible on
    #         FETCH_INTERVAL: 10 # This is the interval the labeler will poll for new images in seconds
```
Note: I'm using version 3.7 of the compose file as I also got an error trying to use version 3.9.I was using 3.8 for a while but then that stopped working too. Not sure if its related.

**Comment by gregordr** [comment]
```
version: "3.7"
services:
    nginx:
        image: imagestored/nginx
        ports:
            - "3000:8080" #Change to X:8080 to access on port X
        depends_on:
            - frontend

    frontend:
        image: imagestored/frontend
        depends_on:
            - backend

    backend:
        image: imagestored/backend
        restart: always
        environment:
            PGSTRING: postgres://postgres:example@db:5432/postgres # This string has to resolve to a Postgres database on version 11
        volumes:
            - ./media/:/code/media/
        depends_on:
            - db

    db:
        image: postgres:11
        restart: always
        environment:
            POSTGRES_PASSWORD: example
        volumes:
            - ./data/:/var/lib/postgresql/data/

    detectron2: # You can use either detectron2, which is slower but more accurate, or yolo, which is faster but has less accurate results, and is ver>         
        image: imagestored/detectron2
        environment:
            BASE_ADDRESS: backend:4000 # Set this to the address the backend will be accessible on
            FETCH_INTERVAL: 10 # This is the interval the labeler will poll for new images in seconds

    # yolo: # You can use either detectron2, which is slower but more accurate, or yolo, which is faster but has less accurate results, and is very le>    #     image: imagestored/yolo
    #     environment:
    #         BASE_ADDRESS: backend:4000 # Set this to the address the backend will be accessible on
    #         FETCH_INTERVAL: 10 # This is the interval the labeler will poll for new images in seconds
```

How about this?

**Comment by Kevr0n228** [comment]
No change unfortunately. everything else seems to be ready to go. I added `POSTGRES_USER: postgres` to `docker-compose`, removed all imagestore containers and volumes, and ran it again but came up with the same. However, I noticed this in the startup logs:
![image](https://user-images.githubusercontent.com/10468194/111881018-fa1fab80-8984-11eb-94e4-c1ca7f32cb66.png)
which might explain why the password is failing. According to [this SO post](https://stackoverflow.com/questions/59715622/docker-compose-and-create-db-in-postgres-on-init), there might be a directory that postgres sees already containing a database but I'm not sure where to look for it.

**Comment by gregordr** [comment]
That's weird. Can you try running the script from an empty directory?

**Comment by Kevr0n228** [comment]
Fortunately I'm running this on a newly added server, so I just removed everything from docker and cleared out the directory I installed it in. While I was doing this, I noticed that there were two postgres images installed. After removing both and re-running the imagestore compose file, it worked without error and I'm able to access and upload images!
![image](https://user-images.githubusercontent.com/10468194/111884707-bc794d80-8999-11eb-87ba-4906cbe3fbe9.png)
Thanks for the help, marking this as closed.
